plutus-benchmark/marlowe/src/PlutusBenchmark/Marlowe/Scripts/Data/RolePayout.hs

```diff
@@ -0,0 +1,108 @@
+
+-- editorconfig-checker-disable-file
+
+
+-----------------------------------------------------------------------------
+--
+-- Module      :  $Headers
+-- License     :  Apache 2.0
+--
+-- Stability   :  Experimental
+-- Portability :  Portable
+--
+-- | Marlowe validators.
+--
+-----------------------------------------------------------------------------
+
+
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
+{-# OPTIONS_GHC -fno-ignore-interface-pragmas #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:defer-errors #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:target-version=1.0.0 #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+
+
+module PlutusBenchmark.Marlowe.Scripts.Data.RolePayout
+  (-- * Payout Validator
+    rolePayoutValidatorHash
+  , rolePayoutValidatorBytes
+  , rolePayoutValidator
+  , mkRolePayoutValidator
+  ) where
+
+
+import PlutusLedgerApi.Data.V2 qualified as Data
+import PlutusLedgerApi.V2 (CurrencySymbol, ScriptHash (..), SerialisedScript, TokenName,
+                           serialiseCompiledCode)
+import PlutusLedgerApi.V2.Data.Contexts qualified as Data
+import PlutusTx (CompiledCode, unsafeFromBuiltinData)
+import PlutusTx.Plugin ()
+import PlutusTx.Prelude as PlutusTxPrelude (Bool (..), BuiltinData, BuiltinUnit, check, toBuiltin,
+                                            ($), (.))
+
+import Cardano.Crypto.Hash qualified as Hash
+import Data.ByteString qualified as BS
+import Data.ByteString.Short qualified as SBS
+import PlutusLedgerApi.V1.Value qualified as Val
+import PlutusTx qualified
+
+
+-- | Tag for the Marlowe payout validator.
+data TypedRolePayoutValidator
+
+
+-- | The Marlowe payout validator.
+mkRolePayoutValidator :: (CurrencySymbol, TokenName)  -- ^ The datum is the currency symbol and role name for the payout.
+                      -> ()                           -- ^ No redeemer is required.
```

Comment:
This should just be `BuiltinData`. Otherwise you need to convert `BuiltinData` to `()`, which is not free.

Reply:
This needs to be done in both Marlowe versions. Opened an issue for https://github.com/IntersectMBO/plutus-private/issues/1468